src/components/Timeout/Timeout.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './timeout.css';
+
+export default function Timeout(props) {
+  //track idle time in background
+  //set time in minutes for idle time before modal should appear, and countdown time before auto logout
+  const idleTime = 1;
+  const countdownTime = 5;
+
+  const [idle, setIdle] = useState(false);
+
+  const time = useRef(0);
+
+  function reset() {
+    time.current = 0;
+  }
+
+  useEffect(() => {
+    let interval = setInterval(() => {
+      time.current += 1;
+      if (time.current === idleTime * 60) {
+        setIdle(true);
+      }
+      console.log('idletime', time.current);
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, [idleTime]);
+
+  return (
+    <div onMouseOver={() => reset()} onKeyPress={() => reset()}>
+      {idle && <Modal countdownTime={countdownTime} setIdle={setIdle} />}
+    </div>
+  );
+}
+
+const Modal = ({ countdownTime, setIdle }) => {
+  //modal with countdown clock
+  const [minutes, setMinutes] = useState(Math.floor(countdownTime));
+  const [seconds, setSeconds] = useState((countdownTime * 60) % 60);
+
+  useEffect(() => {
+    let time = countdownTime * 60;
+
+    const timer =
+      countdownTime > 0 &&
+      setInterval(() => {
+        time -= 1;
+        setMinutes(Math.floor(time / 60));
+        setSeconds(time % 60);
+        if (time === 0) {
+          time = 0;
+          clearInterval(timer);
+        }
+      }, 1000);
+
+    return () => clearInterval(timer);
+  }, [countdownTime]);
+
+  return (
+    <div className="modal-background">
```

Comment:
it looks like we're doing css in js with this repo from what I'm seeing here https://github.com/cfernando415/femmecubator-ui-repo/pull/87/files

Reply:
pretty new to me too but that seems to be hot lately

Reply:
Yeah I see it a lot in React code. I'll change it